allennlp/tests/data/tokenizers/pretrained_transformer_tokenizer_test.py

```diff
@@ -307,9 +307,75 @@ def test_token_idx_sentence_pairs(self):
             == idxs[second_sentence_start_index]
         )
 
+    def test_intra_word_tokenize(self):
+        tokenizer = PretrainedTransformerTokenizer("bert-base-cased")
+
+        sentence = "A, [MASK] AllenNLP sentence.".split(" ")
+        expected_tokens = [
+            "[CLS]",
+            "A",
+            ",",
+            "[MASK]",
+            "Allen",
+            "##NL",
+            "##P",
+            "sentence",
+            ".",
+            "[SEP]",
+        ]
+        expected_offsets = [
+            (1, 2),
+            (3, 3),
+            (4, 6),
+            (7, 8),
+        ]
+        tokens, offsets = tokenizer.intra_word_tokenize(sentence)
+        tokens = [t.text for t in tokens]
+        assert tokens == expected_tokens
+        assert offsets == expected_offsets
+
+        # sentence pair
+        sentence_1 = "A, [MASK] AllenNLP sentence.".split(" ")
+        sentence_2 = "A sentence.".split(" ")
+        expected_tokens = [
+            "[CLS]",
+            "A",
+            ",",
+            "[MASK]",
+            "Allen",
+            "##NL",
+            "##P",
+            "sentence",
+            ".",
+            "[SEP]",
+            "A",
+            "sentence",
+            ".",
+            "[SEP]",
+        ]
+        expected_offsets_a = [
+            (1, 2),
+            (3, 3),
+            (4, 6),
+            (7, 8),
+        ]
+        expected_offsets_b = [
+            (10, 10),
+            (11, 12),
+        ]
+        tokens, offsets_a, offsets_b = tokenizer.intra_word_tokenize_sentence_pair(
```

Comment:
It's a bit cleaner to just have a separate test case for this function.

Reply:
That's what I thought as well but I decided to follow the convention in that test file (https://github.com/allenai/allennlp/blob/4e6c06d8aef916e1f8428dbdba41b9f416353bc9/allennlp/tests/data/tokenizers/pretrained_transformer_tokenizer_test.py#L43).

Reply:
Good point! Local consistency is useful even when the surrounding code is a bit suboptimal.